src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -348,18 +348,94 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
+            from ._client_factory import cf_shared_gallery_image
+            gallery_image_info = cf_shared_gallery_image(cmd.cli_ctx).get(
+                location=location, gallery_unique_name=image_info[0], gallery_image_name=image_info[1])
+        elif is_community_gallery_image_id(gallery_image_reference):
+            # the gallery image reference is community gallery image id
+            from azure.cli.command_modules.vm._vm_utils import parse_community_gallery_image_id
+            image_info = parse_community_gallery_image_id(gallery_image_reference)
+            from ._client_factory import cf_community_gallery_image
+            gallery_image_info = cf_community_gallery_image(cmd.cli_ctx).get(
+                location=location, public_gallery_name=image_info[0], gallery_image_name=image_info[1])
+        else:
+            # the gallery image reference is compute gallery image id
+            from msrestazure.tools import parse_resource_id
+            terms = parse_resource_id(gallery_image_reference)
+            gallery_image_reference_rg, gallery_name, gallery_image_definition = \
+                terms['resource_group'], terms['name'], terms['child_name_1']
+            client = _compute_client_factory(cmd.cli_ctx)
+            gallery_image_info = client.gallery_images.get(resource_group_name=gallery_image_reference_rg,
+                                                           gallery_name=gallery_name,
+                                                           gallery_image_name=gallery_image_definition)
+        if hasattr(gallery_image_info, 'hyper_v_generation'):
+            if gallery_image_info.hyper_v_generation == 'V1':
+                logger.warning('Please consider upgrading security for your VM resources by using Gen 2 OS image and '
+                               'Trusted Launch security type. To know more about Trusted Launch, please visit '
+                               'https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
+            elif gallery_image_info.hyper_v_generation == 'V2':
+                if hyper_v_generation == 'V1':
+                    logger.warning(
+                        'Please consider upgrading security for your VM resources by using Gen 2 OS Disk and '
+                        'Trusted Launch security type. Please use "--hyper-v-generation V2" to set the '
+                        'hypervisor generation as Gen2 for OS disk. To know more about Trusted Launch, '
+                        'please visit https://docs.microsoft.com/en-us/azure/virtual-machines/trusted-launch')
+                # will set default value of hyper_v_generation
+                if hasattr(gallery_image_info, 'features') and gallery_image_info.features \
+                        and 'SecurityType' in gallery_image_info.features \
+                        and gallery_image_info.features['SecurityType'] == 'TrustedLaunchSupported':
```

Comment:
@zhoxing-ms - Filter for gallery image security type can be **TrustedLaunch** or **TrustedLaunchSupported**.

Reply:
> @zhoxing-ms - Filter for gallery image security type can be **TrustedLaunch** or **TrustedLaunchSupported**.

Hi @AjKundnani, is this modification meets your expectation?
![image](https://user-images.githubusercontent.com/105691024/194682337-e0d630fe-d867-44e2-8590-b8ef2e878b2f.png)

Reply:
> > @zhoxing-ms - Filter for gallery image security type can be **TrustedLaunch** or **TrustedLaunchSupported**.
> 
> Hi @AjKundnani, is this modification meets your expectation? ![image](https://user-images.githubusercontent.com/105691024/194683700-212cdb4a-2a7e-45fa-b464-198b2b8b3ea5.png)

@yanzhudd - If we can include **TrustedLaunchAndConfidentialVmSupported** filter, it will ensure future completeness as well.

Reply:
> > > @zhoxing-ms - Filter for gallery image security type can be **TrustedLaunch** or **TrustedLaunchSupported**.
> > 
> > 
> > Hi @AjKundnani, is this modification meets your expectation? ![image](https://user-images.githubusercontent.com/105691024/194683700-212cdb4a-2a7e-45fa-b464-198b2b8b3ea5.png)
> 
> @yanzhudd - If we can include **TrustedLaunchAndConfidentialVmSupported** filter, it will ensure future completeness as well.

@AjKundnani Is this modification meets your expectation?
![image](https://user-images.githubusercontent.com/105691024/194698078-00eeb74a-130f-4e49-b737-233142d2bb75.png)